From 5ae584fd61c5508e3569f97af90c2b4a4c7dd0f5 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Tue, 29 Apr 2025 17:10:46 +0100
Subject: [PATCH 1/2] rc.yml: Stop mangling workdir in python3 macros

The weird workdir managling causes more issues than it solves
---
 ypkg2/rc.yml | 51 ++++++++++++---------------------------------------
 1 file changed, 12 insertions(+), 39 deletions(-)

diff --git a/ypkg2/rc.yml b/ypkg2/rc.yml
index 8fed984..6b778a7 100644
--- a/ypkg2/rc.yml
+++ b/ypkg2/rc.yml
@@ -245,49 +245,22 @@ actions:
         python_compile
     - python3_setup: |
         function python3_setup() {
-                if [[ -e $PKG_BUILD_DIR/.workdir ]]; then
-                    cd "$(cat $PKG_BUILD_DIR/.workdir)"
-                else
-                    echo "$workdir" > $PKG_BUILD_DIR/.workdir
-                fi
-
-                instdir=`basename "$PWD"`
-                pushd ..
-                    cp -a "$instdir" py3build && pushd py3build
-                        if [[ -f "pyproject.toml" ]]; then
-                            python3 -m build --wheel --no-isolation
-
-                        else
-                            echo "No pyproject.toml file found, assuming project isn't PEP517 compatibile"
-                            python3 setup.py build $* || exit
-                        fi
-                    popd
-                popd
+            if [[ -f "pyproject.toml" ]]; then
+                python3 -m build --wheel --no-isolation $*
+            else
+                echo "No pyproject.toml file found, assuming project isn't PEP517 compatibile"
+                python3 setup.py build $* || exit
+            fi
         }
         python3_setup
     - python3_install: |
         function python3_install() {
-                if [[ -e $PKG_BUILD_DIR/.workdir ]]; then
-                    cd "$(cat $PKG_BUILD_DIR/.workdir)"
-                else
-                    echo "$workdir" > $PKG_BUILD_DIR/.workdir
-                fi
-
-                instdir=`basename "$PWD"`
-                pushd ..
-                    if [[ ! -d py3build ]]; then
-                        cp -a "$instdir" py3build
-                    fi
-                    pushd py3build
-                        if [[ -f "pyproject.toml" ]]; then
-                            python3 -m installer --destdir=%installroot% dist/*.whl
-
-                        else
-                            echo "No pyproject.toml file found, installing setuptools"
-                            python3 setup.py install --root="%installroot%" $* || exit
-                        fi
-                    popd
-                popd
+            if [[ -f "pyproject.toml" ]]; then
+                python3 -m installer --destdir=%installroot% dist/*.whl $*
+            else
+                echo "No pyproject.toml file found, installing setuptools"
+                python3 setup.py install --root="%installroot%" $* || exit
+            fi
         }
         python3_install
     - python3_test: |

From 3572d2802af76267253e3d018cd94e31912ea8f9 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Tue, 29 Apr 2025 17:53:56 +0100
Subject: [PATCH 2/2] rc.yml: Adjust python3_test

setup.py test no longer works as a entrypoint with new versions.

Let's use unittest instead.
---
 ypkg2/rc.yml | 19 +------------------
 1 file changed, 1 insertion(+), 18 deletions(-)

diff --git a/ypkg2/rc.yml b/ypkg2/rc.yml
index 6b778a7..edf4048 100644
--- a/ypkg2/rc.yml
+++ b/ypkg2/rc.yml
@@ -265,33 +265,16 @@ actions:
         python3_install
     - python3_test: |
         function python3_test() {
-            if [[ -d py3build ]]; then
-                cd py3build
-            fi
-
             if [[ -z $PYTHONPATH ]]; then
                 export PYTHONPATH=%installroot%/usr/lib/python%python3_version%/site-packages:"$PWD"
-                if [[ -d build/lib ]]; then
-                    PYTHONPATH+=/build/lib
-                fi
                 local do_unset=true
             fi
 
-            if [[ -z $1 ]]; then
-                python3 setup.py test || exit 1
-            elif [[ $1 =~ .*\.py$ ]] || [[ $1 == \-* ]]; then
-                python3 "$@" || exit 1
-            else
-                "$@" || exit
-            fi
+            python3 -m unittest discover
 
             if [[ $do_unset ]]; then
                 unset PYTHONPATH
             fi
-
-            if [[ -d ../py3build ]]; then
-                cd ..
-            fi
         }
         python3_test
     - python3_compile: |
